aws/resource_aws_cloudfront_distribution.go

```diff
@@ -478,6 +478,56 @@ func resourceAwsCloudFrontDistribution() *schema.Resource {
 					},
 				},
 			},
+			"origin_group": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Set:      originGroupHash,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"origin_id": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.NoZeroValues,
+						},
+						"failover_criteria": {
+							Type:     schema.TypeSet,
+							Required: true,
+							Set:      failoverCriteriaHash,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"status_codes": {
+										Type:     schema.TypeList,
```

Comment:
This should be a `TypeSet` if the API does not care about ordering.

```suggestion
										Type:     schema.TypeSet,
```